File: SimulationUtils/hs1d/src_python/Hillslope1D/process/tools/time/TimeUnit.py
class TimeUnit(object):
    """
        Class managing the conversion between time units dor the calculation
        ######################################################################
        @param
          tmax : the maximal time of the timeserie
          unit : string corresponding to the time unit
            choices : year, days, hour, min, sec

        @attributes
          tmax : the maximal time of the timeserie
          unit : string corresponding to the time unit
        ######################################################################
    """

    def __init__(self, tmax=86400*35, unit='sec'):
        self.tmax = tmax
        self.unit = unit


    def time_to_seconds(self,tmax):
        """
            Conversion between each time unit to seconds
        """
        sec_in_min = 60
        sec_in_hour = sec_in_min*60
        sec_in_day = sec_in_hour*24
        sec_in_year = sec_in_day*365

        if self.unit == '':
            self.unit = ('enter time units - seconds, minutes, days, year')
            if self.unit == 'days':
                tmax = tmax*sec_in_day
            elif self.unit == 'hour':
                tmax = tmax*sec_in_hour
            elif self.unit == 'min':
                tmax = tmax*sec_in_min
            elif self.unit == 'sec':
                tmax = tmax
            elif self.unit == 'year':
                tmax = tmax*sec_in_year
            else:
                print('UNIT UNKNOWN #s\n', self.unit)

        elif self.unit == 'year':
            tmax = self.tmax*sec_in_year
        elif self.unit == 'days':
            tmax = tmax * sec_in_day
        elif self.unit == 'hour':
            tmax = tmax * sec_in_hour
        elif self.unit == 'min':
            tmax = tmax * sec_in_min
        elif self.unit == 'sec':
            tmax = tmax
        else:
            print('UNIT UNKNOWN #s\n', self.unit)

        return tmax

    def time_to_days(self):
        sec_in_day = 86400
        min_in_day = 1440
        hour_in_day = 24
        day_in_year = 365


        if self.unit == '':
            self.unit = ('enter time units - seconds, minutes, days, year')
            if self.unit == 'hour':
                self.tmax = self.tmax/hour_in_day
            elif self.unit == 'min':
                self.tmax = self.tmax/min_in_day
            elif self.unit == 'sec':
                self.tmax = self.tmax/sec_in_day
            elif self.unit == 'year':
                self.tmax = self.tmax*day_in_year
            else:
                print('UNIT UNKNOWN #s\n', self.unit)

        elif self.unit == 'sec':
            self.tmax = self.tmax/self.sec_in_day
        elif self.unit == 'min':
            self.tmax = self.tmax / min_in_day
        elif self.unit == 'hour':
            self.tmax = self.tmax / hour_in_day
        elif self.unit == 'year':
            self.tmax = self.tmax*day_in_year
        else:
            print('UNIT UNKNOWN #s\n', self.unit)

        return self.tmax

    def time_to_years(self):
        days_in_year = 365
        hours_in_year = 24*days_in_year
        min_in_year = 60*hours_in_year
        sec_in_year = 60*min_in_year


        if self.unit == '':
            self.unit = ('enter time units - seconds, minutes, days, year')
            if self.unit == 'days':
                self.tmax = self.tmax*days_in_year
            elif self.unit == 'hour':
                self.tmax = self.tmax*hours_in_year
            elif self.unit == 'min':
                self.tmax = self.tmax*min_in_year
            elif self.unit == 'sec':
                self.tmax = self.tmax
            else:
                print('UNIT UNKNOWN #s\n', self.unit)

        elif self.unit == 'sec':
            self.tmax = self.tmax / sec_in_year
        elif self.unit == 'min':
            self.tmax = self.tmax / min_in_year
        elif self.unit == 'hour':
            self.tmax = self.tmax / hours_in_year
        elif self.unit == 'days':
            self.tmax = self.tmax / days_in_year
        else:
            print('UNIT UNKNOWN #s\n', self.unit)

        return self.tmax

    def time_to_hours(self):
        hours_in_day = 24
        hours_in_year = 365*hours_in_day
        min_in_hour = 60
        sec_in_hour = 60*min_in_hour

        if self.unit == '':
            self.unit = ('enter time units - seconds, minutes, days, year')
            if self.unit == 'year':
                self.tmax = self.tmax*hours_in_year
            elif self.unit == 'days':
                self.tmax = self.tmax*hours_in_day
            elif self.unit == 'hour':
                self.tmax = self.tmax
            elif self.unit == 'min':
                self.tmax = self.tmax/min_in_hour
            elif self.unit == 'sec':
                self.tmax = self.tmax/sec_in_hour
            else:
                print('UNIT UNKNOWN #s\n', self.unit)

        elif self.unit == 'sec':
            self.tmax = self.tmax / sec_in_hour
        elif self.unit == 'min':
            self.tmax = self.tmax / min_in_hour
        elif self.unit == 'hour':
            self.tmax = self.tmax
        elif self.unit == 'days':
            self.tmax = self.tmax * hours_in_day
        elif self.unit == 'year':
            self.tmax = self.tmax * hours_in_year
        else:
            print('UNIT UNKNOWN #s\n', self.unit)

        return self.tmax

File: SimulationUtils/hs1d/src_matlab/process/tools/jr_tools/time/time_unit.m
classdef time_unit
    % time units
    properties(Access=protected)
        % unit of time
        unit
    end
    
    methods(Access=public)
        % constructor
        function obj=time_unit(unit)
            if(nargin<1); obj.unit='sec'; else obj.unit=unit; end
        end
    end

    methods(Static)
        
        function t=time_to_years(t,time_unit)
            %transforms travel times in time_unit to years
            days_in_year = 365;
            hours_in_year = 24 * days_in_year;
            mins_in_year = 60 * hours_in_year;
            secs_in_year = 60 * mins_in_year;
            
            if (strcmp (time_unit, 'sec'))
                t(:) = t(:)/ secs_in_year;
            elseif (strcmp (time_unit, 'min'))
                t(:) = t(:)/ mins_in_year;
            elseif (strcmp (time_unit, 'hour'))
                t(:) = t(:)/ hours_in_year;
            elseif (strcmp (time_unit, 'day'))
                t(:) = t(:)/ days_in_year;
            elseif (strcmp (time_unit, 'year'))
                t(:) = t(:)/ 1 ;
            else
                fprintf('UNIT UNKNOWN %s\n', time_unit); 
            end
        end
        
        function t=time_to_seconds(t,time_unit)
            %transforms travel times in time_unit to seconds
            secs_in_mins = 60;
            secs_in_hours = 60 * secs_in_mins;
            secs_in_days = 24 * secs_in_hours;
            secs_in_year = 365 * secs_in_days;
            
            if (strcmp (time_unit, 'year'))
                t(:) = t(:) * secs_in_year;
            elseif (strcmp (time_unit, 'day'))
                t(:) = t(:) * secs_in_days;
            elseif (strcmp (time_unit, 'hour'))
                t(:) = t(:) * secs_in_hours;
            elseif (strcmp (time_unit, 'min'))
                t(:) = t(:) * secs_in_mins;
            elseif (strcmp (time_unit, 'sec'))
                t(:) = t(:) * 1 ;
            else
                fprintf('UNIT UNKNOWN %s\n', unit); 
            end
        end
        
        function t=time_to_days(t,time_unit)
            %transforms travel times in time_unit to days
            days_in_year = 365;
            hours_in_day = 24;
            mins_in_day = 60 * hours_in_day;
            secs_in_day = 60 * mins_in_day;
            
            if (strcmp (time_unit, 'sec'))
                t(:) = t(:)/ secs_in_day;
            elseif (strcmp (time_unit, 'min'))
                t(:) = t(:)/ mins_in_day;
            elseif (strcmp (time_unit, 'hour'))
                t(:) = t(:)/ hours_in_day;
            elseif (strcmp (time_unit, 'day'))
                t(:) = t(:) / 1;
            elseif (strcmp (time_unit, 'year'))
                t(:) = t(:)* days_in_year ;
            else
                fprintf('UNIT UNKNOWN %s\n', time_unit); 
            end
        end
    end


end